src/Microsoft.OData.Core/UriParser/Parsers/CountSegmentParser.cs

```diff
@@ -82,8 +82,8 @@ public CountSegmentToken CreateCountSegmentToken(QueryToken countedInstance)
                 while (this.lexer.CurrentToken.Kind != ExpressionTokenKind.CloseParen)
                 {
                     string text = this.UriQueryExpressionParser.EnableCaseInsensitiveBuiltinIdentifier
-                        ? this.lexer.CurrentToken.Text.ToLowerInvariant()
-                        : this.lexer.CurrentToken.Text;
+                        ? this.lexer.CurrentToken.Text.ToString().ToLowerInvariant()
+                        : this.lexer.CurrentToken.Text.ToString();

```

Comment:
Is it necessary to call `ToString()` here? Since we are only using the strings for switch matching and not returning it, maybe we can avoid the extra allocations. Maybe we can use `Text.Equals(...)` instead of the `switch` statement. And for the `$` prefix, maybe we can also perform a comparison without having to do the string concatenation.

Reply:
Thanks. Improved it.